packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -434,10 +436,12 @@ export function getInternalReactConstants(
         return 'Lazy';
       case MemoComponent:
       case SimpleMemoComponent:
-        return (
-          (elementType && elementType.displayName) ||
-          (type && type.displayName) ||
-          getDisplayName(resolvedType, 'Anonymous')
+        // Display name in React does not use `Memo` as a wrapper but fallback name.
```

Comment:
I think we should sync the display names for memo as well. React will computed `Component` while devtools will display it as `Memo(Component)`. No strong opinion whether we want to use `Memo` as a wrapper or fallback. For me personally `React.memo` is used like a higher-order component i.e. `Memo` as a wrapper is intuitive for me.